addon-manager/impl/src/main/java/org/jboss/forge/addon/manager/impl/ui/AddonUpdateCommand.java

```diff
@@ -0,0 +1,148 @@
+   }
+
+   @Override
+   public void initializeUI(UIBuilder builder) throws Exception
+   {
+      Project project = Projects.getSelectedProject(projectFactory, builder.getUIContext());
+      if (project != null)
+      {
+         MetadataFacet facet = project.getFacet(MetadataFacet.class);
+         Coordinate c = facet.getOutputDependency().getCoordinate();
+         named.setDefaultValue(AddonId.from(c.getGroupId() + ":" + c.getArtifactId(), c.getVersion())
+                  .toCoordinates());
+      }
+
+      named.setCompleter(new UICompleter<String>()
+      {
+         @Override
+         public Iterable<String> getCompletionProposals(UIContext context, InputComponent<?, String> input, String value)
+         {
+            Set<String> items = new TreeSet<String>();
+            Set<Addon> addons = furnace.getAddonRegistry().getAddons();
+            for(Addon addon : addons) {
+               items.add(addon.getId().getName());
+            }
+            return items;
+         }
+      });
+
+      named.addValidator(new UIValidator()
+      {
+         @Override
+         public void validate(UIValidationContext context)
+         {
+            String coordinate = (String) context.getCurrentInputComponent().getValue();
+            try
+            {
+               AddonId addonId = CoordinateUtils.resolveCoordinate(coordinate,Versions.getSpecificationVersionFor(this),resolver);
+               System.out.println(addonId);
+               Set<Addon> addons = furnace.getAddonRegistry().getAddons();
+               boolean match =false;
+               for(Addon addon : addons) {
+                  if(addon.getId().getName().equals(addonId.getName())) {
+                     match=true;
+                     break;
+                  }
+               }
+               if(!match) {
+                  context.addValidationError(context.getCurrentInputComponent(), "\"" + coordinate
+                           + "\" does not refer to any installed Addon coordinate");
+               }
+            }
+            catch (IllegalArgumentException e)
+            {
+               context.addValidationError(context.getCurrentInputComponent(), "\"" + coordinate
+                        + "\" is not a valid Addon coordinate");
+            }
+         }
+      });
+
+      builder.add(named);
+   }
+
+   @Override
+   public Result execute(UIExecutionContext context)
+   {
+      AddonId addonId = CoordinateUtils.resolveCoordinate(named.getValue(),Versions.getSpecificationVersionFor(this),resolver);
+      AddonId maxAddonId = addonId;
+      try
+      {
+         Response<AddonId[]> resolveVersions = resolver.resolveVersions(addonId.getName());
+         
+         for(AddonId id : resolveVersions.get()) {
+             if(id.getVersion().compareTo(maxAddonId.getVersion()) > 0 ) {
+                maxAddonId = id;
+             }
+         }
+         addonManager.install(maxAddonId).perform();
```

Comment:
Also addonManager.install() returns an object stating what will change when perform() is called, it would be nice to display that too and maybe prompt for execution